core/sql/optimizer/BindRI.cpp

```diff
@@ -417,12 +450,18 @@ void RefConstraint::getPredicateText(NAString &text,
 				     NAString *corrName) const
 {
   NAString tblText = ( (corrName == NULL) ? tblName.getQualifiedNameAsAnsiString() : *corrName);
+  int pos= 0;
   text += "(";
   for (CollIndex i = 0; i < keyColumns.entries(); i++)
     {
-      if (i)
+      if(isHiddenColumn(keyColumns[i]->getColName()) )
```

Comment:
Wonder if we can use isSystemColumn() instead. SystemColumn includes SYSKEY column also. Also there is a method isSysKeyColumn(), just in case you want to allow SYSKEY COLUMN.